test/silhouette.jl

```diff
@@ -34,4 +34,21 @@ c = [3, 1]
 
 @test silhouettes(a, c, D) ≈ [0.5, 0.5, -1/3, 0.0]
 
+@testset "silhouette() errors when degenerated clustering is given" begin
```

Comment:
Actually, you have added two test sets. The first one is
```suggestion
@testset "silhouettes() handle zero cluster distances correctly" begin
```
(note that the clustering is not degenerated).